kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/utils/TestRow.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.utils
+
+import scala.collection.JavaConverters._
+
+import io.delta.kernel.data.Row
+import io.delta.kernel.types._
+
+/**
+ * Corresponding Scala class for each Kernel data type:
+ * - BooleanType --> boolean
+ * - ByteType --> byte
+ * - ShortType --> short
+ * - IntegerType --> int
+ * - LongType --> long
+ * - FloatType --> float
+ * - DoubleType --> double
+ * - StringType --> String
+ * - DateType --> int (number of days since the epoch)
+ * - TimestampType --> long (number of microseconds since the unix epoch)
+ * - DecimalType --> java.math.BigDecimal
+ * - BinaryType --> Array[Byte]
+ *
+ * TODO: complex types
+ * - StructType?
+ * - ArrayType?
+ * - MapType?
+ */
+class TestRow(val values: Array[Any]) {
+  // TODO: we could make this extend Row and create a way to generate Seq(Any) from Rows but it
+  //  would complicate a lot of the code for not much benefit
```

Comment:
What's the benefit of generating `Seq[Any]`?
What's the harm of not being able to do so now?

Reply:
> What's the benefit of generating Seq[Any]?

To compare Rows we need a way to turn them into iterables. So either
1) Convert all read Rows --> TestRows (which can be converted to iterable) and then compare with TestRows created in tests
2) Make TestRow extend Row. Convert both read Rows and TestRows to iterable and then compare

> What's the harm of not being able to do so now?

Differences if we did this
- Create rows in tests with `Row(0, 1, ...)` instead of `TestRow(0, 1, ...)` (minor difference)
- `TestRow` needs to implement all methods in `Row` (annoying)
- Instead of converting `Row` --> `TestRow` and then comparing, we'd have some other method to go from `Row` --> iterable (pretty equivalent)
  - Note we'd still use `TestRow` throughout since we need to create rows in `prepareRow` etc

Reply:
Slightly lost my train of thought on this PR since it's been a while; going to double check this tomorrow